parachain/primitives/core/src/operating_mode.rs

```diff
@@ -0,0 +1,82 @@
+use codec::{Decode, Encode, FullCodec, MaxEncodedLen};
+use frame_support::{
+	dispatch::DispatchResult, traits::EnsureOrigin, PalletError, RuntimeDebug, StorageValue,
+};
+use scale_info::TypeInfo;
+use sp_std::fmt::Debug;
+
+/// Error generated by the `OwnedBridgeModule` trait.
+#[derive(Encode, Decode, TypeInfo, PalletError)]
+pub enum BridgeModuleError {
+	/// All pallet operations are halted.
+	Halted,
+}
+
+/// Operating mode for a bridge module.
+pub trait OperatingMode: Send + Copy + Debug + FullCodec {
```

Comment:
Lets rename this to `OperatingModeTrait`, as it allows to simplify the name of `BasicOperatingMode` enum, which is part of the API.

```suggestion
pub trait OperatingModeTrait: Send + Copy + Debug + FullCodec {
```